### PR TITLE
Feature/add log file

### PR DESCRIPTION
**Pull request by iancardosozup**
**- What I did**
Add global flag --log-path or -l to set log file path 
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -183,6 +187,54 @@ func (c *Config) SetLogLevel(logLevel string) {
...
+			return nil, err
+		}
+	} else if _, err = os.Stat(logPath); os.IsNotExist(err) {
+		return nil, err
+	}
+	return createLogFile(logPath)
+}
+
+func createLogFile(dir string) (*os.File, error) {
+	fileName := dir + "/horusec-log-" + time.Now().Format("2006-01-02 15:04:05") + ".log"
```
Comment (matheusalcantarazup) [inline config/config.go:215]: We should use `filepath.Join(dir, "horusec-log", ...)` instead. Concatenating paths with strings will broke on Windows.

File: cmd/app/start/start_test.go [hunk 2]
```diff
@@ -360,14 +355,26 @@ func TestStartCommand_Execute(t *testing.T) {
...
 		cobraCmd.SetArgs([]string{"-p", "./", "-o", "json", "-O", "./tmp-json.json"})
 
 		assert.NoError(t, cobraCmd.Execute())
-		outputBytes, err := ioutil.ReadAll(stdoutMock)
-		output := string(outputBytes)
+		err := w.Close()
 		assert.NoError(t, err)
+		os.Stdout = oldStdout
+		output := <-outC
+
```
Comment (matheusalcantarazup) [inline cmd/app/start/start_test.go:377]: This seems to be a hack to me, I think that we can change the stdout of cli without changing the global `os.Stdout` value.

File: config/config.go [hunk 3]
```diff
@@ -183,6 +186,59 @@ func (c *Config) SetLogLevel(logLevel string) {
...
+func (c *Config) SetLogFilePath(logFilePath string) {
+	c.logFilePath = logFilePath
+}
+
+func (c *Config) SetLogOutput() error {
+	file, err := getFile(c.logFilePath)
+	if err != nil {
+		return err
+	}
+	logger.LogSetOutput(file, os.Stdout)
```
Comment (matheusalcantarazup) [inline config/config.go:198]: I think that `os.Stdout`value should be a parameter, so we can test without override the global `os.Stdout` value.

File: cmd/app/start/start.go [hunk 4]
```diff
@@ -132,6 +132,10 @@ func (s *Start) setConfig(startCmd *cobra.Command) {
 	s.configs = s.configs.NewConfigsFromViper().NormalizeConfigs()
 	s.configs = s.configs.NewConfigsFromEnvironments().NormalizeConfigs()
 	s.configs = s.configs.NewConfigsFromCobraAndLoadsCmdStartFlags(startCmd).NormalizeConfigs()
+	err := s.configs.SetLogOutput()
```
Comment (matheusalcantarazup) [inline cmd/app/start/start.go:135]:
```
if err := s.configs.SetLogOutput(); err != nil {
        logger.LogErrorWithLevel(messages.MsgErrorSettingLogFile, err)
}
```

File: config/config.go [hunk 5]
```diff
@@ -183,6 +186,59 @@ func (c *Config) SetLogLevel(logLevel string) {
...
+		if err != nil {
+			return nil, err
+		}
+	}
+	fileName := filepath.Join(dirPath, "horusec-log-"+time.Now().Format("2006-01-02 15:04:05")+".log")
+	file, err := os.Create(fileName)
+	if err != nil {
+		return nil, err
+	}
+	return file, nil
```
Comment (matheusalcantarazup) [inline config/config.go:229]:
```
return os.Create(filename)
```

File: config/config.go [hunk 6]
```diff
@@ -183,6 +186,59 @@ func (c *Config) SetLogLevel(logLevel string) {
...
+}
+func getDirName() (string, error) {
+	ex, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	return ex, nil
+}
```
Comment (matheusalcantarazup) [inline config/config.go:237]:
```
return os.Getwd()
```

File: config/config_test.go [hunk 7]
```diff
@@ -452,3 +464,61 @@ func TestConfig_ToBytes(t *testing.T) {
...
+
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "no such file or directory")
+	})
+	t.Run("Should fail when log path is a file", func(t *testing.T) {
+		config := &Config{}
+		file, err := os.Create("./test.txt")
+		if err != nil {
+			t.Error(err)
+		}
```
Comment (matheusalcantarazup) [inline config/config_test.go:482]: Use assert.Nil or require.Nil instead.

File: config/config.go [hunk 8]
```diff
@@ -183,6 +188,48 @@ func (c *Config) SetLogLevel(logLevel string) {
...
+func (c *Config) SetLogFilePath(logFilePath string) {
+	c.logFilePath = logFilePath
+}
+
+func (c *Config) SetLogOutput(writers ...io.Writer) error {
+	file, err := c.getFile(c.logFilePath)
+	if err != nil {
+		return err
+	}
+	logger.LogSetOutput(append(writers, file)...)
```
Comment (matheusalcantarazup) [inline config/config.go:200]:
```
logger.LogSetOutput(writers, file)
```

**Review by nathanmartinszup — APPROVED** [review]

**Pull request merged**